From 8b4d70e611afba5b636b838bc48a912df38282a8 Mon Sep 17 00:00:00 2001
From: Tejun Heo <tj@kernel.org>
Date: Wed, 28 Feb 2024 11:25:58 -1000
Subject: [PATCH 151/153] scx: Move tools/sched_ext/Kconfig content into
 sched-ext.rst

Kconfig files are usually used to describe config options and
tools/sched_ext/Kconfig isn't being referenced from anywhere. Let's move its
content into the documentation.
---
 Documentation/scheduler/sched-ext.rst | 15 ++++++++++++++-
 tools/sched_ext/Kconfig               |  9 ---------
 2 files changed, 14 insertions(+), 10 deletions(-)
 delete mode 100644 tools/sched_ext/Kconfig

diff --git a/Documentation/scheduler/sched-ext.rst b/Documentation/scheduler/sched-ext.rst
index e3847b44e..b01c4f22f 100644
--- a/Documentation/scheduler/sched-ext.rst
+++ b/Documentation/scheduler/sched-ext.rst
@@ -22,7 +22,20 @@ Switching to and from sched_ext
 ===============================
 
 ``CONFIG_SCHED_CLASS_EXT`` is the config option to enable sched_ext and
-``tools/sched_ext`` contains the example schedulers.
+``tools/sched_ext`` contains the example schedulers. The following config
+options should be enabled to use sched_ext:
+
+.. code-block:: none
+
+    CONFIG_BPF=y
+    CONFIG_SCHED_CLASS_EXT=y
+    CONFIG_BPF_SYSCALL=y
+    CONFIG_BPF_JIT=y
+    CONFIG_DEBUG_INFO_BTF=y
+    CONFIG_BPF_JIT_ALWAYS_ON=y
+    CONFIG_BPF_JIT_DEFAULT_ON=y
+    CONFIG_PAHOLE_HAS_SPLIT_BTF=y
+    CONFIG_PAHOLE_HAS_BTF_TAG=y
 
 sched_ext is used only when the BPF scheduler is loaded and running.
 
diff --git a/tools/sched_ext/Kconfig b/tools/sched_ext/Kconfig
deleted file mode 100644
index 6543fcf19..000000000
--- a/tools/sched_ext/Kconfig
+++ /dev/null
@@ -1,9 +0,0 @@
-CONFIG_BPF=y
-CONFIG_SCHED_CLASS_EXT=y
-CONFIG_BPF_SYSCALL=y
-CONFIG_BPF_JIT=y
-CONFIG_DEBUG_INFO_BTF=y
-CONFIG_BPF_JIT_ALWAYS_ON=y
-CONFIG_BPF_JIT_DEFAULT_ON=y
-CONFIG_PAHOLE_HAS_SPLIT_BTF=y
-CONFIG_PAHOLE_HAS_BTF_TAG=y
-- 
2.43.0.232.ge79552d197

